tests/restore_test.go

```diff
@@ -0,0 +1,683 @@
+			})
+
+			AfterEach(func() {
+				deleteSnapshot(snapshot)
+				if webhook != nil {
+					deleteWebhook(webhook)
+				}
+			})
+
+			It("should successfully restore", func() {
+				var origSpec *v1.VirtualMachineSpec
+
+				Eventually(func() bool {
+					var updatedVM *v1.VirtualMachine
+					vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					origSpec = vm.Spec.DeepCopy()
+					Expect(origSpec.Template.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]).To(Equal(resource.MustParse("64M")))
+
+					vm.Spec.Template.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("128M")
+					updatedVM, err = virtClient.VirtualMachine(vm.Namespace).Update(vm)
+					if errors.IsConflict(err) {
+						return false
+					}
+					vm = updatedVM
+					Expect(err).ToNot(HaveOccurred())
+					Expect(vm.Spec.Template.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]).To(Equal(resource.MustParse("128M")))
+					return true
+				}, 180*time.Second, time.Second).Should(BeTrue())
+
+				restore := createRestoreDef(vm, snapshot.Name)
+
+				restore, err = virtClient.VirtualMachineRestore(vm.Namespace).Create(restore)
+				Expect(err).ToNot(HaveOccurred())
+
+				restore = waitRestoreComplete(restore, vm)
+				Expect(restore.Status.Restores).To(HaveLen(0))
+				Expect(restore.Status.DeletedDataVolumes).To(HaveLen(0))
+
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+				Expect(vm.Spec).To(Equal(*origSpec))
+
+				deleteRestore(restore)
+			})
+
+			It("should reject restore if VM running", func() {
+				patch := []byte("[{ \"op\": \"replace\", \"path\": \"/spec/running\", \"value\": true }]")
+				vm, err := virtClient.VirtualMachine(vm.Namespace).Patch(vm.Name, types.JSONPatchType, patch)
+				Expect(err).ToNot(HaveOccurred())
+
+				restore := createRestoreDef(vm, snapshot.Name)
+
+				restore, err = virtClient.VirtualMachineRestore(vm.Namespace).Create(restore)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("VirtualMachine %q is running", vm.Name)))
+			})
+
+			It("should reject restore if another in progress", func() {
+				fp := admissionregistrationv1beta1.Fail
+				whPath := "/foobar"
+				whName := "dummy-webhook-deny-vm-update.kubevirt.io"
+				wh := &admissionregistrationv1beta1.ValidatingWebhookConfiguration{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "temp-webhook-deny-vm-update",
+					},
+					Webhooks: []admissionregistrationv1beta1.ValidatingWebhook{
+						{
+							Name:          whName,
+							FailurePolicy: &fp,
+							Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+								Operations: []admissionregistrationv1beta1.OperationType{
+									admissionregistrationv1beta1.Update,
+								},
+								Rule: admissionregistrationv1beta1.Rule{
+									APIGroups:   []string{"kubevirt.io"},
+									APIVersions: []string{"v1alpha3", "v1"},
```

Comment:
this likely will change at some point. I'd recommend using `virtv1.ApiSupportedWebhookVersions` here as it will automatically get updated as new api versions are added.